plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestIgniteConnectorTest.java

```diff
@@ -0,0 +1,391 @@
+        try (TestTable testTable = simpleTable()) {
+            assertThatThrownBy(() -> query(format("SELECT * FROM TABLE(system.query(query => 'INSERT INTO %s VALUES (3, 4)'))", testTable.getName())))
+                    .hasMessage("line 1:21: Table function system.query not registered");
+            assertQuery("SELECT * FROM " + testTable.getName(), "VALUES (1, 1), (2, 2)");
+        }
+    }
+
+    @Override
+    protected TestTable simpleTable()
+    {
+        return new TestTable(onRemoteDatabase(), format("%s.simple_table", getSession().getSchema().orElseThrow()), "(col BIGINT, id bigint primary key)", ImmutableList.of("1, 1", "2, 2"));
+    }
+
+    @Override
+    public void testNativeQueryIncorrectSyntax()
+    {
+        // table function disabled for Ignite, because it doesn't provide ResultSetMetaData, so the result relation type cannot be determined
+        assertThatThrownBy(() -> query("SELECT * FROM TABLE(system.query(query => 'some wrong syntax'))"))
+                .hasMessage("line 1:21: Table function system.query not registered");
+    }
+
+    @Override
+    protected void verifyAddNotNullColumnToNonEmptyTableFailurePermissible(Throwable e)
+    {
+        // Ignite will not throw any exception here.
+    }
+
+    @Override
+    public void testCharVarcharComparison()
+    {
+        // Ignite will map char to varchar, skip
+        throw new SkipException("Ignite map char to varchar, skip test");
+    }
+
+    @Override
+    protected String errorMessageForInsertIntoNotNullColumn(String columnName)
+    {
+        return format("Failed to insert data: Null value is not allowed for column '%s'", columnName.toUpperCase(Locale.ENGLISH));
+    }
+
+    @Override
+    public void testCharTrailingSpace()
+    {
+        throw new SkipException("Ignite not support char trailing space");
+    }
+
+    @Override
+    public void testDropAndAddColumnWithSameName()
+    {
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_drop_add_column", "(x int NOT NULL, y int, z int)", ImmutableList.of("1,2,3"))) {
+            assertUpdate("ALTER TABLE " + table.getName() + " DROP COLUMN y");
+            assertQuery("SELECT * FROM " + table.getName(), "VALUES (1, 3)");
+
+            assertUpdate("ALTER TABLE " + table.getName() + " ADD COLUMN y int");
+            // Follow the behavior of Ignite
+            assertQuery("SELECT * FROM " + table.getName(), "VALUES (1, 3, 2)");
+        }
+    }
+
+    @Override
+    protected Optional<DataMappingTestSetup> filterDataMappingSmokeTestData(DataMappingTestSetup dataMappingTestSetup)
+    {
+        String typeName = dataMappingTestSetup.getTrinoTypeName();
+        if (typeName.equals("timestamp(3) with time zone") ||
+                typeName.equals("timestamp(6) with time zone")) {
+            return Optional.of(dataMappingTestSetup.asUnsupported());
+        }
+
+        if (typeName.startsWith("time")) {
+            return Optional.empty();
```

Comment:
Please leave a comment why it uses `Optional.empty()` instead of `asUnsupported()`.

Reply:
Update the check to `asUnsupported()` and remove the redundant check for `timestamp(x) with time zone` above